src/Raygun4php/Transports/GuzzleSync.php

```diff
@@ -37,10 +46,26 @@ public function transmit(RaygunMessageInterface $message): bool
             $httpResponse = $this->httpClient
                                  ->request('POST', '/entries', ['body' => $messageJson]);
         } catch (TransferException $th) {
+            $this->logger->error($th->getMessage(), json_decode($messageJson, true));
             return false;
         }
 
         $responseCode = $httpResponse->getStatusCode();
-        return $responseCode === 202;
+
+        if ($responseCode !== 202) {
+            $logMsg = "Expected response code 202 but got {$responseCode}";
+            if ($responseCode < 400) {
+                $this->logger->warning($logMsg, json_decode($messageJson, true));
+            } else {
+                $this->logger->error($logMsg, json_decode($messageJson, true));
+            }
+            return false;
+        }
+        return true;
+    }
+
+    public function setLogger(LoggerInterface $logger)
```

Comment:
You have a void return type on the async equivalent of this class. Do you want to add it here too?

Reply:
Done!